freshmaker/lightblue.py

```diff
@@ -1825,3 +1830,50 @@ def get_fixed_published_image(self, name, version, image_group, rpm_nvrs, conten
         image = images[0]
         image.resolve(self)
         return image
+
+    def is_latest_eus_image(self, image):
+        """
+        Check if the image is EUS base image and his repository marked with
+        auto_rebuild tag. EUS repository contains '-els' suffix.
+
+        :param ContainerImage image: image to check
+        :param dict repo_to_auto_rebuild_tags: mapping of processed repositories
+            names to set of their 'auto_rebuild_tags'
+        :return: True if image is EUS base image with 'auto_rebuild_tag', False otherwise
+        """
+        def get_repo(repository):
+            query = {
+                "objectType": "containerRepository",
+                "query": {
+                    "$and": [
+                        {
+                            "field": "repository",
+                            "op": "=",
+                            "rvalue": repository
+                        },
+                    ]
+                },
+                "projection": [
+                    {"field": "auto_rebuild_tags", "include": True, "recursive": True}
+                ]
+            }
+            repos = self.find_container_repositories(query, auto_rebuild=True)
+            return repos[0]
+
+        for repo in image["repositories"]:
+            repo_name = repo["repository"]
+            # Check tags only if it's EUS image
+            if re.match(r".*-els($|\W)", repo_name):
+                auto_rebuild_tags = self.repo_to_auto_rebuild_tags.get(repo_name, None)
+                if auto_rebuild_tags is None:
+                    auto_rebuild_tags = get_repo(repo_name)["auto_rebuild_tags"]
+                    self.repo_to_auto_rebuild_tags[repo_name] = auto_rebuild_tags
+
+                if not auto_rebuild_tags:
+                    continue
+                repo_tags = (tag["name"] for tag in repo["tags"])
+                for tag in repo_tags:
+                    if tag in auto_rebuild_tags:
```

Comment:
Optional: You can take this syntax if you prefer
```suggestion
                repo_tags = {tag["name"] for tag in repo["tags"]}
                if repo_tags & auto_rebuild_tags:
```

Reply:
Wont it consume more time? Because in my option it will return when encounter a first tag. And in your option it needs to make intersection of two sets.

Reply:
You are right. I thought it was more readable what I suggested, but you can keep it as is. I also forgot that `auto_rebuild_tags` is not a set.

If you change `auto_rebuild_tags` to be a set, the most efficient way would be the following:
```suggestion
                for tag in repo["tags"]:
                    if tag["name"] in auto_rebuild_tags:
```

This avoids essentially looping through `repo["tags"]` and makes the `in` operation O(1) instead of O(n).

Reply:
Done